# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/dialog/nice_scroll.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./menubox.c.orig	2003-03-09 12:09:44.000000000 +0100
+++ ./menubox.c	2003-03-09 12:24:36.000000000 +0100
@@ -259,6 +259,8 @@
 	scrollamt = choice - (max_choice - 1);
 	choice = max_choice - 1;
     }
+    while ( (scrollamt+max_choice < item_no) &&
+	(choice > max_choice/2) ) { scrollamt++; choice--; }
 
     /* Print the menu */
     for (i = 0; i < max_choice; i++)
